sql/sql_insert.cc

```diff
@@ -830,51 +832,45 @@ bool Sql_cmd_insert::mysql_insert(THD *thd,TABLE_LIST *table_list)
     if (error <= 0 || thd->get_transaction()->cannot_safely_rollback(
         Transaction_ctx::STMT))
     {
-//TIANMU UPGRADE BEGIN
-#if defined(TIANMU)
       bool tianmu_engine = insert_table->s->db_type() ? insert_table->s->db_type()->db_type == DB_TYPE_TIANMU: false;
-      if (mysql_bin_log.is_open()&& !tianmu_engine)
-#else
-      if (mysql_bin_log.is_open())
-#endif
-//END
+      if (mysql_bin_log.is_open()&& !(tianmu_engine && tianmu_sysvar_insert_delayed))
```

Comment:
suggest: you add some words on `tianmu_sysvar_insert_delayed`. why we add this var in.